embeddings/src/lib.rs

```diff
@@ -38,97 +37,203 @@ fn handle_embeddings(req: Request) -> Result<Response> {
+            let embedding_records: Vec<Embedding> = conn
+                .execute(query, &[])?
+                .rows()
+                .map(|row| -> anyhow::Result<Embedding> { row.try_into() })
+                .collect::<anyhow::Result<Vec<Embedding>>>()?;
+
+            trace!("Rows: {:?}", embedding_records);
+            Ok(http::Response::builder()
+                .status(http::StatusCode::OK)
+                .body(Some(serde_json::to_string(&embedding_records)?.into()))
+                .unwrap())
+        }
+    }
 }
 
-fn create_embeddings(_req: Request, _params: Params) -> Result<Response> {
-    // Serialize the body of the req to Vec<embedding>
-    // Create a SQL query to search for existing and redact
-    // Submit the delta to SQL
+fn create_embeddings(req: Request, _params: Params) -> Result<Response> {
+    let embedding_request: EmbeddingRequest = serde_json::from_slice(
+        req.body()
+            .as_ref()
+            .map(|b| -> &[u8] { b })
+            .unwrap_or_default(),
+    )
+    .unwrap();
 
-    let my_embedding = Embedding {
-        id: 0,
-        reference: String::from("My ref"),
-        text: String::from("My text"),
-        embedding: Some(vec![1.23, 4.56, 7.89]),
-    };
-    let vec = json!(my_embedding.embedding);
-    let blob = serde_json::to_vec(&vec)?;
-
-    json!(my_embedding.embedding);
+    let embeddings = generate_and_store_embedding(embedding_request);
 
-    // let b_vec = serde_json::from_value(vec)?;
+    trace!("Result: {:?}", embeddings);
 
-    let query = "INSERT INTO embeddings (reference, text, embedding) VALUES(?, ?, ?) RETURNING id;";
-    let query_params = [
-        sqlite::ValueParam::Text(my_embedding.reference.as_str()),
-        sqlite::ValueParam::Text(my_embedding.text.as_str()),
-        sqlite::ValueParam::Blob(blob.as_slice()),
-    ];
-    let conn = Connection::open_default()?;
-    let result = conn.execute(query, &query_params)?;
-    info!("Result: {:?}", result.rows[0]);
     Ok(http::Response::builder()
         .status(http::StatusCode::CREATED)
         .body(None)
         .unwrap())
 }
 
-fn delete_embeddings(_req: Request, _params: Params) -> Result<Response> {
-    // Delete the record with id
+fn delete_embeddings(_req: Request, params: Params) -> Result<Response> {
+    let conn = Connection::open_default()?;
+    let query_params = [sqlite::ValueParam::Integer(
+        params.get("id").unwrap().parse().unwrap(),
+    )];
+    let _ = conn.execute("DELETE FROM embeddings WHERE id = (?)", &query_params);
 
+    // Report appropriate status code - e.g., if delete fails
     Ok(http::Response::builder()
         .status(http::StatusCode::OK)
         .body(None)
         .unwrap())
 }
 
-#[derive(Debug, Serialize, Deserialize)]
-struct Embedding {
-    id: u32,
-    reference: String,
-    text: String,
-    embedding: Option<Vec<f32>>,
+fn generate_and_store_embedding(embedding_req: EmbeddingRequest) -> Result<Vec<Embedding>> {
```

Comment:
I found this function very hard to read probably because of the use of `embeddings` for both the request and the result of the call to the AI model. 

Also, I don't understand why you're mutating the request. It seems like you have `EmbeddingRequest.embedding` set to None and then make the request to `generate_embeddings` and set the `EmbeddingRequest.embedding` to the result of that request. Why? It seems like the only thing you do with the result of this function is print it out to logs so this seems more complicated than it needs to be?